NetscapeBookmarkParser.php

```diff
@@ -206,19 +216,58 @@ public function parseString($bookmarkString) {
      * @return int Unix timestamp corresponding to a successfully parsed date,
      *             else current date and time
      */
-    public static function parseDate($date)
+    public function parseDate($date)
     {
         if (strtotime('@'.$date)) {
             // Unix timestamp
+            if ($this->normalizeDates) {
+                $date = $this->normalizeDate($date);
+            }
             return strtotime('@'.$date);
         } else if (strtotime($date)) {
             // attempt to parse a known compound date/time format
             return strtotime($date);
         }
         // current date & time
-        return time();
+        return $time;
+    }
+
+    /**
+     * Normalizes a date by supposing it is comprised in a given range
+     *
+     * Although most bookmarking services return dates formatted as a Unix epoch
+     * (seconds elapsed since 1970-01-01 00:00:00) or human-readable strings,
+     * some services return microtime epochs (microseconds elapsed since
+     * 1970-01-01 00:00:00.000000) WITHOUT using a delimiter for the microseconds
+     * part...
+     *
+     * This is likely to raise issues in the distant future!
+     *
+     * @see https://stackoverflow.com/questions/33691428/datetime-with-microseconds
+     * @see https://stackoverflow.com/questions/23929145/how-to-test-if-a-given-time-stamp-is-in-seconds-or-milliseconds
+     * @see https://stackoverflow.com/questions/539900/google-bookmark-export-date-format
+     * @see https://www.wired.com/2010/11/1110mars-climate-observer-report/
+     *
+     * @param string $epoch     Unix timestamp to normalize
+     *
+     * @return string Unix timestamp in seconds, within the expected range
+     */
+    public function normalizeDate($epoch) {
+        $now = new \DateTime();
+        $range = new \DateInterval('P'.$this->dateDeltaYears.'Y');
+        $maxDate = $now->add($range);
```

Comment:
For simple intervals, you can just use `new \DateTime('+30 years')`.

Reply:
nice, this is also more user-friendly!